project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc

```diff
@@ -19,57 +19,79 @@ $plugin = array(
  * Render callback.
  */
 function c4m_features_homepage_my_groups_content_type_render($subtype, $conf, $args, $context) {
-  global $user;
+  $groups = c4m_features_homepage_get_user_groups();
 
-  $account = user_load($user->uid);
-  $first_groups = '';
-  $extra_groups = '';
-
-  // Get All user's groups (We get them sorted by id).
-  $groups = og_get_groups_by_user($account, NULL, TRUE);
-
-  if (empty($groups['node'])) {
-    return FALSE;
+  if (empty($groups)) {
+    return;
   }
 
-  $nids = array_values($groups['node']);
-
-  $my_groups = node_load_multiple($nids);
-
-  usort($my_groups, 'c4m_sort_object_by_title');
-
-  // Render node views for each group,
-  // First 3 groups are displayed (first_groups), the rest are hidden and only
-  // shown when user clicks on "Show all" (extra_groups).
+  $rendered_groups = '';
   $count = 0;
-  foreach ($my_groups as $group) {
-    if ($group->type != 'group') {
-      continue;
-    }
+  // Render node views for each group,
+  foreach ($groups as $group) {
     $node_view = node_view($group, 'block_list');
-    if ($count < 3) {
-      $first_groups .= drupal_render($node_view);
-    }
-    else {
-      $extra_groups .= drupal_render($node_view);
-    }
+    $rendered_groups .= drupal_render($node_view);
+
     $count++;
+    if ($count >= 5) {
+      break;
+    }
   }
 
+  $link_options = array(
+    'query' => array(
+      'filter' => 'groups',
+    )
+  );
   $variables = array(
-    'first_groups' => $first_groups,
-    'extra_groups' => $extra_groups,
-    'show_all_link' => $count > 3 ? TRUE : FALSE,
+    'groups' => $rendered_groups,
+    'link' => url('groups', $link_options),
   );
 
   $block = new stdClass();
   $block->module = 'c4m_features_homepage';
   $block->title = '';
-  $block->content = $first_groups ? theme('c4m_features_homepage_my_groups', $variables) : FALSE;
+  $block->content = $rendered_groups ? theme('c4m_features_homepage_my_groups', $variables) : FALSE;
   return $block;
 }
 
 /**
+ * Returns the list of groups of an user.
+ *
+ * @param object $account.
+ *   User object. Defaults global user if NULL.
+ *
+ * @returns mixed|null
+ *   List of groups or NULL if the user has no groups.
+ */
+function c4m_features_homepage_get_user_groups($account = NULL) {
+  $groups = c4m_user_og_get_groups($account);
+  usort($groups, 'c4m_sort_object_by_title');
+
+  return $groups;
+}
+
+/**
+ * Returns the list of groups of an user, grouped by the membership type.
+ *
+ * @param object $account.
+ *   User object. Defaults global user if NULL.
+ *
+ * @returns mixed|null
+ *   List of groups or NULL if the user has no groups.
+ */
+function c4m_features_homepage_get_user_groups_grouped_by_membership($account = NULL) {
```

Comment:
Functionality seems nowhere used.
Please also make sure owner is on top > followed by admin > member > membership requested.

Ref:

show on the home page
- groups I own ("owner")
- groups I manage ("admnistrator")
- groups I'm member of ("member")
- groups I requested membership for ("membership requested")

"my archived groups" do not need to be shown on home page
ordering

1st: owner > administrator > member > membership requested
2nd: alphabetical